frontend/packages/console-app/src/components/nav/ClusterMenu.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react';
+import {
+  Button,
+  Divider,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateSecondaryActions,
+  Menu,
+  MenuContent,
+  MenuGroup,
+  MenuInput,
+  MenuItem,
+  MenuList,
+  TextInput,
+  Title,
+} from '@patternfly/react-core';
+import fuzzysearch from 'fuzzysearch';
+import { useTranslation } from 'react-i18next';
+import { useActivePerspective } from '@console/dynamic-plugin-sdk';
+import { useActiveCluster, usePerspectiveExtension } from '@console/shared';
+import { ACM_PERSPECTIVE_ID } from '../../consts';
+import ClusterMenuToggle from './ClusterMenuToggle';
+
+const ClusterCIcon: React.FC = () => <span className="co-m-resource-icon">C</span>;
+
+const NoResults: React.FC<{
+  onClear: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+}> = ({ onClear }) => {
+  const { t } = useTranslation();
+  return (
+    <EmptyState>
+      <Title size="md" headingLevel="h4">
+        {t('console-app~No cluster found')}
+      </Title>
+      <EmptyStateBody>{t('console-app~No results match the filter criteria.')}</EmptyStateBody>
+      <EmptyStateSecondaryActions>
+        <Button variant="link" onClick={onClear}>
+          {t('console-app~Clear filter')}
+        </Button>
+      </EmptyStateSecondaryActions>
+    </EmptyState>
+  );
+};
+
+const ClusterFilter: React.FC<{
+  filterRef: React.Ref<any>;
+  onFilterChange: (filterText: string) => void;
+  filterText: string;
+}> = ({ filterText, filterRef, onFilterChange }) => {
+  const { t } = useTranslation();
+  return (
+    <MenuInput translate="no">
+      <TextInput
+        autoFocus
+        placeholder={t('console-app~Find a cluster...')}
+        aria-label={t('console-app~Find a cluster...')}
+        iconVariant="search"
+        type="search"
+        value={filterText}
+        onChange={onFilterChange}
+        ref={filterRef}
+      />
+    </MenuInput>
+  );
+};
+
+const ClusterGroup: React.FC<{
+  clusters: ClusterMenuItem[];
+}> = ({ clusters }) => {
+  const [activeCluster] = useActiveCluster();
+
+  return clusters.length === 0 ? null : (
+    <MenuGroup translate="no" label="Clusters">
+      <MenuList>
+        {clusters.map((cluster) => (
+          <MenuItem
+            translate="no"
+            key={cluster.key}
+            itemId={cluster.key}
+            isSelected={activeCluster === cluster.key}
+            onClick={(e) => {
+              e.preventDefault();
+              cluster.onClick();
+            }}
+          >
```

Comment:
data-test-id="cluster-dropdown-item"

add this for e2e tests to access the menu item. if you want a better name, then the test id used here should match the one in nav.ts...ie cluster-menu-item 

cc: @TheRealJon